### PR TITLE
dialects: (affine) add store and load operations

### PR DESCRIPTION
**Pull request by superlopuh**
@Groverkss

### CODE REVIEW

File: xdsl/dialects/affine.py [hunk 1]
```diff
@@ -83,6 +88,50 @@ def from_region(
...
 
+@irdl_op_definition
+class Store(IRDLOperation):
+    name = "affine.store"
+
+    T = Annotated[Attribute, ConstraintVar("T")]
+
+    value = operand_def(T)
+    memref = operand_def(MemRefType[T])
+    map = opt_attr_def(AffineMapAttr)
```
Comment (webmiche) [inline xdsl/dialects/affine.py:99]: I am confused, why is there this map, whereas the MLIR docu suggests an `indices`-field?

Comment (webmiche) [reply xdsl/dialects/affine.py:99]: Did this change to the most recent MLIR version?

**Review by webmiche — COMMENTED** [review]
interop tests? 😉

File: tests/filecheck/dialects/affine/affine_ops.mlir [hunk 2]
```diff
@@ -30,4 +30,19 @@
     // CHECK-NEXT:   "affine.yield"(%{{.*}}) : (!test.type<"int">) -> ()
     // CHECK-NEXT: }) {"lower_bound" = -10 : index, "upper_bound" = 10 : index, "step" = 1 : index} : (!test.type<"int">) -> !test.type<"int">
 
+    %memref = "test.op"() : () -> memref<2x3xf64>
+    %value = "test.op"() : () -> f64
+    "affine.store"(%value, %memref) {"map" = affine_map<() -> (0, 0)>} : (f64, memref<2x3xf64>) -> ()
```
Comment (Groverkss) [inline tests/filecheck/dialects/affine/affine_ops.mlir:35]: Isn't the map wrong? Shouldn't it be affine_map<(d0, d1) -> (0, 0)>?

File: xdsl/dialects/affine.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/dialects/affine.py:99]: It seems so, I added these in such a way to parse the output of the version of MLIR I built on my machine.

Comment (webmiche) [reply xdsl/dialects/affine.py:99]: I see :+1: 
Once the interop tests works, we will know if this aligns with our CI version :sweat_smile:

File: tests/filecheck/dialects/affine/affine_ops.mlir (hunk 2 above)
Comment (superlopuh) [reply tests/filecheck/dialects/affine/affine_ops.mlir:35]: When I change it it throws an error. But also it doesn't surprise me, what would those values be?

File: xdsl/dialects/affine.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/dialects/affine.py:99]: I added an interop test, it passes. I guess one more reason to update to a more recent MLIR

Comment (webmiche) [reply xdsl/dialects/affine.py:99]: Fair, though we will need to add property support for that, which we wanted to postpone as long as possible...

**Pull request merged**